### PR TITLE
chore: allow to use underscore as identifier

### PR DESCRIPTION
**Pull request by ghost**
# Description

allow to use underscore as identifier

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Can you add a test for this?

**Comment by jfecher** [comment]
@kevaundray looks like there's a small test in the parser added testing let bindings with an underscore.

**Review by jfecher — CHANGES_REQUESTED** [review]
Thanks for the PR! Underscores not being an identifier is something I often run into that can be annoying.

As for the fix, instead of mapping an underscore token to an Identifier afterward, I think we should just remove the underscore token entirely. A quick grep shows it is unused, and a single underscore should be counted as an identifier anyway. It looks like we just need to remove the underscore case in `Keyword::lookup_keyword`

File: crates/noirc_frontend/src/hir/resolution/resolver.rs [hunk 1]
```diff
@@ -204,6 +204,8 @@ impl<'a> Resolver<'a> {
         warn_if_unused: bool,
         definition: DefinitionKind,
     ) -> HirIdent {
+        let allow_shadowing = allow_shadowing || name.eq("_");
```
Comment (jfecher) [inline crates/noirc_frontend/src/hir/resolution/resolver.rs:207]:
```suggestion
        let allow_shadowing = allow_shadowing || &name == "_";
```

**Review by jfecher — APPROVED** [review]
Thanks!

**Pull request merged**